src/Bicep.IO/FileSystem/FileSystemFileHandle.cs

```diff
@@ -34,15 +44,25 @@ public IDirectoryHandle GetParent()
             return new FileSystemDirectoryHandle(this.FileSystem, parentDirectoryPath);
         }
 
-        public Stream OpenRead() => this.FileSystem.File.OpenRead(Uri.GetFileSystemPath());
+        public Stream OpenRead() => this.FileSystem.File.OpenRead(this.Uri.GetFileSystemPath());
 
         public Stream OpenWrite()
         {
             this.GetParent().EnsureExists();
 
-            return this.FileSystem.File.OpenWrite(Uri.GetFileSystemPath());
+            return this.FileSystem.File.OpenWrite(this.Uri.GetFileSystemPath());
+        }
+
+        public void Delete() => this.FileSystem.File.Delete(this.Uri.GetFileSystemPath());
+
+        public void MakeExecutable()
+        {
+            if (!OperatingSystem.IsWindows())
+            {
+                this.FileSystem.File.SetUnixFileMode(this.Uri.GetFileSystemPath(), UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.UserExecute);
```

Comment:
>UserWrite [](http://example.com/codeflow?start=120&length=9)

Is UserWrite needed to mark it as executable?

Reply:
It's not, but chmod 700 seems to be common. Also, the original code has the `UserWrite` flag, so keeping it just in case.